camel/functions/search_functions.py

```diff
@@ -156,24 +228,14 @@ def text_extract_from_web(url: str) -> str:
         str: All texts extract from the web.
     """
     import requests
-    from bs4 import BeautifulSoup
+    from newspaper import Article
 
     try:
         # Request the target page
-        response_text = requests.get(url).text
-
-        # Parse the obtained page
-        soup = BeautifulSoup(response_text, features="html.parser")
-
-        for script in soup(["script", "style"]):
-            script.extract()
-
-        text = soup.get_text()
-        # Strip text
-        lines = (line.strip() for line in text.splitlines())
-        chunks = (phrase.strip() for line in lines
-                  for phrase in line.split("  "))
-        text = ".".join(chunk for chunk in chunks if chunk)
+        article = Article(url)
+        article.download()
+        article.parse()
+        text = article.text.replace("\n", "")
```

Comment:
Why replace `\n` by `""`?

Reply:
What is considered here is to convert the parsed text into continuous text, and then provide it to the summarize_text function as knowledge.